swiftnav/src/reference_frame.rs

```diff
@@ -0,0 +1,656 @@
+            s: 0.0,
+            s_dot: 0.0,
+            rx: 0.0,
+            rx_dot: 0.0,
+            ry: 0.0,
+            ry_dot: 0.0,
+            rz: 0.0,
+            rz_dot: 0.0,
+            epoch: 2010.0,
+        };
+        let initial_velocity = super::ECEF::default();
+        let position = super::ECEF::default();
+
+        let transformed_velocity = params.transform_velocity(&initial_velocity, &position);
+        assert_float_eq!(transformed_velocity.x(), 0.1, abs_all <= 1e-4);
+        assert_float_eq!(transformed_velocity.y(), 0.2, abs_all <= 1e-4);
+        assert_float_eq!(transformed_velocity.z(), 0.3, abs_all <= 1e-4);
+    }
+
+    #[test]
+    fn helmert_velocity_scaling() {
+        let params = super::TimeDependentHelmertParams {
+            tx: 0.0,
+            tx_dot: 0.0,
+            ty: 0.0,
+            ty_dot: 0.0,
+            tz: 0.0,
+            tz_dot: 0.0,
+            s: 1.0 / super::TimeDependentHelmertParams::SCALE_SCALE,
+            s_dot: 0.1 / super::TimeDependentHelmertParams::SCALE_SCALE,
+            rx: 90.0,
+            rx_dot: 0.0,
+            ry: 0.0,
+            ry_dot: 0.0,
+            rz: 0.0,
+            rz_dot: 0.0,
+            epoch: 2010.0,
+        };
+        let initial_velocity = super::ECEF::default();
+        let position = super::ECEF::new(1., 2., 3.);
+
+        let transformed_velocity = params.transform_velocity(&initial_velocity, &position);
+        assert_float_eq!(transformed_velocity.x(), 0.1, abs_all <= 1e-4);
+        assert_float_eq!(transformed_velocity.y(), 0.2, abs_all <= 1e-4);
+        assert_float_eq!(transformed_velocity.z(), 0.3, abs_all <= 1e-4);
+    }
+
+    #[test]
+    fn helmert_velocity_rotations() {
+        let params = super::TimeDependentHelmertParams {
+            tx: 0.0,
+            tx_dot: 0.0,
+            ty: 0.0,
+            ty_dot: 0.0,
+            tz: 0.0,
+            tz_dot: 0.0,
+            s: 0.0,
+            s_dot: 0.0,
+            rx: 1.0 / super::TimeDependentHelmertParams::ROTATE_SCALE,
+            rx_dot: 0.1 / super::TimeDependentHelmertParams::ROTATE_SCALE,
+            ry: 2.0 / super::TimeDependentHelmertParams::ROTATE_SCALE,
+            ry_dot: 0.2 / super::TimeDependentHelmertParams::ROTATE_SCALE,
+            rz: 3.0 / super::TimeDependentHelmertParams::ROTATE_SCALE,
+            rz_dot: 0.3 / super::TimeDependentHelmertParams::ROTATE_SCALE,
+            epoch: 2010.0,
+        };
+        let initial_velocity = super::ECEF::default();
+        let position = super::ECEF::new(4., 5., 6.);
+
+        let transformed_velocity = params.transform_velocity(&initial_velocity, &position);
+        assert_float_eq!(transformed_velocity.x(), -0.3, abs_all <= 1e-4);
+        assert_float_eq!(transformed_velocity.y(), 0.6, abs_all <= 1e-4);
+        assert_float_eq!(transformed_velocity.z(), -0.3, abs_all <= 1e-4);
+    }
+
+    #[test]
+    fn helmert_invert() {
```

Comment:
Might be worth adding a full roundtrip test either to this test or a separate one. To ensure that a transformation or all transformations result in the input parameters after two invocations to `.invert()`

Reply:
Added this!